apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/integration-permissions-section.client.tsx

```diff
@@ -3,12 +3,13 @@ import { RadioGroup, RadioGroupItem } from "@/components/ui/radio-group";
 import { cn } from "@/lib/utils";
 import { useState } from "react";
 import { toast } from "sonner";
+import invariant from "tiny-invariant";
 import type { Ecosystem } from "../../../types";
 import { useUpdateEcosystem } from "../../hooks/use-update-ecosystem";
 
 export function IntegrationPermissionsSection({
   ecosystem,
-}: { ecosystem: Ecosystem }) {
+}: { ecosystem?: Ecosystem }) {
```

Comment:
how come rendering this without an ecosystem existing is possible? Feels like that should not be a legal state?

Reply:
Yeah I didn't love this either. Basically need some way to indicate loading state but without duplicating the entire component's structure into a loading component. Also didn't want to just show a spinner since some UI like the new ecosystem button doesn't rely on the async ecosystem fetch

Reply:
probably makes sense to setup a `loading.tsx` for this?

Reply:
The fetch its reliant on is client-side though so would that do anything?